pkg/providers/nutanix/template.go

```diff
@@ -435,3 +439,30 @@ func generateNoProxyList(clusterSpec *cluster.Spec) []string {
 
 	return noProxyList
 }
+
+func generateNutanixFailureDomains(FailureDomains []v1alpha1.NutanixDatacenterFailureDomain) []capxv1beta1.NutanixFailureDomain {
+	var failureDomains []capxv1beta1.NutanixFailureDomain
+	for _, fd := range FailureDomains {
+
+		subnets := []capxv1beta1.NutanixResourceIdentifier{}
+		for _, subnet := range fd.Subnets {
+			subnets = append(subnets, capxv1beta1.NutanixResourceIdentifier{
+				Type: capxv1beta1.NutanixIdentifierType(subnet.Type),
+				Name: subnet.Name,
+				UUID: subnet.UUID,
+			})
+		}
+
+		failureDomains = append(failureDomains, capxv1beta1.NutanixFailureDomain{
+			Name: fd.Name,
+			Cluster: capxv1beta1.NutanixResourceIdentifier{
+				Type: capxv1beta1.NutanixIdentifierType(fd.Cluster.Type),
+				Name: fd.Cluster.Name,
+				UUID: fd.Cluster.UUID,
+			},
+			Subnets:      subnets,
+			ControlPlane: true,
```

Comment:
should this be fd.ControlPlane?

Reply:
yeah what does `ControlPlane: true` signify?

Reply:
ControlPlane: true means the failure domain will be used for control plane VMs and etcd VMs. Th requested feature is to spread Control Plane VMs across several PE's so we're setting CAPX ControlPlane option to true.